### PR TITLE
fix: copy `.eot`, `.otf`, `.ttf`, `.woff`, and `woff2` font files when bundling

### PR DESCRIPTION
**Pull request by eltigerchino**
closes https://github.com/sveltejs/kit/pull/11219
fixes https://github.com/sveltejs/kit/issues/11220

Copies `.ttf` and `.woff` font files when bundling edge functions since those are the most common font files included with packages. Alternatively, we could use the same list of assets as [Vite](https://github.com/vitejs/vite/blob/22b299429599834bf1855b53264a28ae5ff8f888/packages/vite/src/node/optimizer/esbuildDepPlugin.ts#L25-L48).

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

File: packages/adapter-cloudflare/index.js [hunk 1]
```diff
@@ -90,7 +90,9 @@ export default function (options = {}) {
 					format: 'esm',
 					bundle: true,
 					loader: {
-						'.wasm': 'copy'
+						'.wasm': 'copy',
+						'.woff': 'copy',
```
Comment (benmccann) [inline packages/adapter-cloudflare/index.js:94]: is there also a `.woff2` extension we need to add here? I just did a quick search for woff and thought I saw something about that

**Comment by benmccann** [comment]
do we also need to update here? https://github.com/sveltejs/kit/blob/657e52aed75f359c27a6fe53eb4b44c9d107a2f0/packages/adapter-vercel/index.js#L137

File: packages/adapter-cloudflare/index.js (hunk 1 above)
Comment (eltigerchino) [reply packages/adapter-cloudflare/index.js:94]: Yes, I should add `.woff2` as well.

**Comment by eltigerchino** [comment]
> do we also need to update here?
> 
> https://github.com/sveltejs/kit/blob/657e52aed75f359c27a6fe53eb4b44c9d107a2f0/packages/adapter-vercel/index.js#L137

Yes, that's being updated. It's one of the steps required to get `@vercel/og` working since that includes a `.woff` fallback font in the package.

File: packages/adapter-cloudflare-workers/index.js [hunk 2]
```diff
@@ -97,7 +97,10 @@ export default function ({ config = 'wrangler.toml', platformProxy = {} } = {})
 					alias: Object.fromEntries(compatible_node_modules.map((id) => [id, `node:${id}`])),
 					format: 'esm',
 					loader: {
-						'.wasm': 'copy'
+						'.wasm': 'copy',
```
Comment (benmccann) [inline packages/adapter-cloudflare-workers/index.js:100]: it looks like eot and otf are the other two coming up in my searches

Comment (eltigerchino) [reply packages/adapter-cloudflare-workers/index.js:100]: Alright, I've added `.eot` and `.otf` too. Just curious, what kind of search are you conducting?

Comment (benmccann) [reply packages/adapter-cloudflare-workers/index.js:100]: i'd done a good search for "esbuild loader font" or something like that

**Review by benmccann — APPROVED** [review]

**Comment by eltigerchino** [comment]
So apparently, this won't do anything until https://github.com/evanw/esbuild/issues/795 is resolved.
Currently:
* Importing fonts without Vite's help causes a reference error.
* Importing fonts with Vite's help will transform it into a constant so esbuild will never copy it anyway as it's no longer an import statement.

**Comment by benmccann** [comment]
Should we revert this or just leave it?

**Comment by eltigerchino** [comment]
I might have to look at how the manoco editor is referencing their font because it seems esbuild picks it up, which caused the "no loader" error in https://github.com/sveltejs/kit/issues/11220 we should check to see if the reproducible works after upgrading the adapter

**Comment by RonB** [comment]
Hi @eltigerchino,  just curious what the status is, the issue is closed but i cant figure out how to build my app on cloudflare with the moncao editor as it still gives the loader error on the ttf font.
Greatly appreciate your response

**Comment by eltigerchino** [comment]
> Hi @eltigerchino, just curious what the status is, the issue is closed but i cant figure out how to build my app on cloudflare with the moncao editor as it still gives the loader error on the ttf font. Greatly appreciate your response

Hi @RonB . I'd suggest opening a thread in the Svelte Discord or in the GitHub discussions. Closed PRs or issues are not a great place for opening new discussions.

**Pull request merged**